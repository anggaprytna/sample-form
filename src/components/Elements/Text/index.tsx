import React from 'react';
import {Text} from 'react-native';
import {RFValue as fs} from 'react-native-responsive-fontsize';

import styles from './styles';

type PropsText = {
  children: any;
  color?: string;
  style?: Object;
  size?: number;
  type: 'thin' | 'regular' | 'semibold' | 'bold';
};

const Typography: React.FC<PropsText> = props => {
  const TextStyles: any[] = [
    props.type === 'thin' && styles.thin,
    props.type === 'regular' && styles.regular,
    props.type === 'semibold' && styles.semibold,
    props.type === 'bold' && styles.bold,
  ];
  const colors: object = {
    color: props.color ? props.color : '#072c5a',
  };
  const sizeText: object = {
    fontSize: props.size && fs(props.size),
  };
  return (
    <Text
      {...props}
      style={[styles.default, TextStyles, colors, sizeText, props.style]}>
      {props.children}
    </Text>
  );
};

export default Typography;
