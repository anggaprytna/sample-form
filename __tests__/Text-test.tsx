/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Text from '../src/components/Elements/Text';

it('Text', () => {
  // test props
  let TextPrimary = renderer.create(<Text type="bold">asd</Text>);
  let TextDark = renderer.create(<Text type="regular">asd</Text>);
  let TextLight = renderer.create(<Text type="semibold">asd</Text>);
  let textThin = renderer.create(<Text type="thin">asd</Text>);
  expect({TextPrimary, TextDark, TextLight, textThin}).toMatchSnapshot();
});
