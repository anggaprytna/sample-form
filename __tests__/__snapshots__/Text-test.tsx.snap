// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text 1`] = `
Object {
  "TextDark": <Text
    style={
      Array [
        Object {
          "fontFamily": "Gilroy-Light",
          "fontSize": 16,
        },
        Array [
          false,
          Object {
            "fontSize": 16,
          },
          false,
          false,
        ],
        undefined,
      ]
    }
  >
    asd
  </Text>,
  "TextLight": <Text
    style={
      Array [
        Object {
          "fontFamily": "Gilroy-Light",
          "fontSize": 16,
        },
        Array [
          false,
          false,
          Object {
            "fontSize": 16,
          },
          false,
        ],
        undefined,
      ]
    }
  >
    asd
  </Text>,
  "TextPrimary": <Text
    style={
      Array [
        Object {
          "fontFamily": "Gilroy-Light",
          "fontSize": 16,
        },
        Array [
          false,
          false,
          false,
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          },
        ],
        undefined,
      ]
    }
  >
    asd
  </Text>,
  "textThin": <Text
    style={
      Array [
        Object {
          "fontFamily": "Gilroy-Light",
          "fontSize": 16,
        },
        Array [
          Object {
            "fontSize": 16,
          },
          false,
          false,
          false,
        ],
        undefined,
      ]
    }
  >
    asd
  </Text>,
}
`;
